### PR TITLE
[Tests] Improve test_attention_outputs

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

I noticed several vision models overwrite `test_attention_outputs`, but we have a `has_attentions` flag exactly for this purpose.

Hence, I've adapted the test in `test_modeling_common.py` and `test_modeling_tf_common.p`y to improve this.

### CODE REVIEW

File: tests/test_modeling_common.py [hunk 1]
```diff
@@ -554,6 +554,9 @@ def test_training_gradient_checkpointing(self):
             loss.backward()
 
     def test_attention_outputs(self):
+        if not self.has_attentions:
+            return
```
Comment (ydshieh) [inline tests/test_modeling_common.py:558]:
```suggestion
            self.skipTest(reason="This model does not output attentions")
```

File: tests/test_modeling_tf_common.py [hunk 2]
```diff
@@ -830,6 +830,9 @@ def test_keyword_and_dict_args(self):
             self.assertLess(np.sum(np.abs(output_dict - output_keywords)), 1e-6)
 
     def test_attention_outputs(self):
+        if not self.has_attentions:
+            return
```
Comment (ydshieh) [inline tests/test_modeling_tf_common.py:834]:
```suggestion
            self.skipTest(reason="This model does not output attentions")
```

File: tests/test_modeling_common.py (hunk 1 above)
Comment (ydshieh) [reply tests/test_modeling_common.py:558]: We can probably get the model class here, and use f-string to make the reason more explicit. I am a bit lazy here though.

**Review by ydshieh — APPROVED** [review]
Thanks! Yeah, it's better, but we should skip the tests instead of just `return` - so we can see they are skipped in the CI report.

File: tests/test_modeling_common.py (hunk 1 above)
Comment (NielsRogge) [reply tests/test_modeling_common.py:558]: Ok, not that there are many other tests just using "return", so we can update those as well

Comment (NielsRogge) [reply tests/test_modeling_common.py:558]: We can not really get the model class, only the model tester

Comment (ydshieh) [reply tests/test_modeling_common.py]: We can: `BertModelTest` -> `BertModel`, or even look into `self.all_model_classes`. There will definitely a few edge cases to deal with though.

Comment (NielsRogge) [reply tests/test_modeling_common.py]: Ok!

**Review by sgugger — APPROVED** [review]
Nice cleanup, thanks!
